ibis/backends/tests/test_client.py

```diff
@@ -202,3 +203,56 @@ def test_separate_database(con, alternate_current_database, current_data_db):
     db = con.database(current_data_db)
     assert db.name == current_data_db
     assert tmp_db.name == alternate_current_database
+
+
+@pytest.mark.only_on_backends(
+    ['sqlite', 'postgres', 'mysql'],
+    reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_from_dataframe(con, df):
+    drop = 'DROP TABLE IF EXISTS temporary_alltypes'
+    create = (
+        'CREATE TABLE IF NOT EXISTS '
+        'temporary_alltypes AS functional_alltypes'
+    )
+
+    con.raw_sql(drop)
+    con.raw_sql(create)
+
+    temporary = con.table('temporary_alltypes')
+    records = df[:10]
+
+    assert len(temporary.execute()) == 0
+    con.insert('temporary_alltypes', data_obj=records)
+
+    temporary_post_insert = con.table('temporary_alltypes')
+
+    tm.assert_frame_equal(temporary_post_insert.execute(), records)
+
+
+@pytest.mark.only_on_backends(
+    ['sqlite', 'postgres', 'mysql'],
+    reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_from_table(con, from_table_name):
```

Comment:
Do we have a fixture `from_table_name`? Surprising if that's the case.

Reply:
I couldn't find the orchestrator for these testing functions but I was assuming the user will be providing it (`from_table_name` parameter) so that we can insert data from that table.

Reply:
Those arguments are a pytest magic named fixtures. You define a function `con` that generates one or multiple connections, then you register it as a fixture, and pytest executes the test with the return of the function (it possibly executes it multiple times).

In this case there is no user for the test, is all automated. What I would do is to create two rows manually, and then you create a dataframe for the first test, and you can insert them to a temporary table with `raw_sql` in the second.

Reply:
Updated the tests, please take a look.